backend-rust/src/graphql_api.rs

```diff
@@ -3346,6 +3559,56 @@ impl SearchResult {
     }
 }
 
+fn decode_value_with_schema(
+    opt_schema: &Option<Type>,
+    schema_name: &str,
+    value: &Vec<u8>,
+    value_name: &str,
+) -> String {
+    match opt_schema {
```

Comment:
Make this `match` into
```rust
let Some(schema) = opt_schema else {
    return format!(
        "Failed to deserialize {} with {} schema: {:?}",
        value_name,
        schema_name,
        e.display(true)
    )
};
```
Then you avoid indenting the whole rest of the body and the return is not so far from the relevant condition.

Alternatively if this function doesn't work for `None` anyway, maybe just don't take an option but take a `Type` directly? Let the caller decide how to handle it if it is missing.